src/Input/MarkdownInput.php

```diff
@@ -38,13 +38,21 @@ final class MarkdownInput implements MarkdownInputInterface
      */
     private $lineCount;
 
-    public function __construct(string $content)
+    /**
+     * @var int
+     *
+     * @psalm-readonly
+     */
+    private $lineOffset = 0;
+
+    public function __construct(string $content, int $lineOffset = 0)
     {
         if (! \mb_check_encoding($content, 'UTF-8')) {
             throw new UnexpectedEncodingException('Unexpected encoding - UTF-8 or ASCII was expected');
         }
 
-        $this->content = $content;
+        $this->content    = $content;
+        $this->lineOffset = $lineOffset;
```

Comment:
There's no `getLineOffset` method which could be quite useful on its own (accurate debug tracing)

Reply:
Lines only get numbered when iterating over them via `getLines()` which does utilize that variable.